codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/serde/SerializeStructGenerator.kt

```diff
@@ -463,15 +480,26 @@ open class SerializeStructGenerator(
      * input.fooBlobMap.forEach { (key, value) -> entry(key, value.encodeBase64String()) }
      * ```
      */
-    private fun renderBlobEntry(keyShape: Shape, nestingLevel: Int, listMemberName: String) {
+    private fun renderBlobEntry(keyShape: Shape, nestingLevel: Int, listMemberName: String, isSparse: Boolean) {
         val containerName = if (nestingLevel == 0) "input." else ""
         val (keyName, valueName) = keyValueNames(nestingLevel)
         val keyValue = keyValue(keyShape, keyName)
 
-        writer.write(
-            "$containerName$listMemberName.forEach { ($keyName, $valueName) -> entry($keyValue, $valueName.#T()) }",
-            RuntimeTypes.Core.Text.Encoding.encodeBase64String,
-        )
+        when (isSparse) {
+            true -> {
+                writer.withBlock("$containerName$listMemberName.forEach { ($keyName, $valueName) ->", "}") {
+                    writer.withBlock("if (value != null) {", "} else entry(key, null as String?)") {
+                        writer.write("entry($keyValue, $valueName.#T())", RuntimeTypes.Core.Text.Encoding.encodeBase64String)
+                    }
+                }
+            }
+            false -> {
+                writer.write(
+                    "$containerName$listMemberName.forEach { ($keyName, $valueName) -> entry($keyValue, $valueName.#T()) }",
+                    RuntimeTypes.Core.Text.Encoding.encodeBase64String,
+                )
+            }
+        }
```

Comment:
possible simplification using `wrapBlockIf`:

```kt
writer.withBlock("$containerName$listMemberName.forEach { ($keyName, $valueName) ->", "}") {
    writer.wrapBlockIf(isSparse, "if ($valueName != null) {", "} else entry($keyName, null as String?)") {
        writer.write("entry($keyValue, $valueName.#T())", RuntimeTypes.Core.Text.Encoding.encodeBase64String)
    } 
}
```

Reply:
It's a good simplification but it would cause codegen changes to the serializers that are not sparse when not necessary

Reply:
I'm not seeing the codegen change, unless you are you talking about the newline?

Non-sparse currently:
```kt
$containerName$#listMemberName.forEach { ($keyName, $valueName) -> entry($keyValue, $valueName.encodeBase64String()) }
```

Non-sparse with my suggestio:
```kt
$containerName$listMemberName.forEach { ($keyName, $valueName) ->
    entry($keyValue, $valueName.encodeBase64String())
}
```

Reply:
Yeah I'm talking about the new line